sympy/matrices/common.py

```diff
@@ -1278,7 +1278,7 @@ def _eval_is_zero_matrix(self):
         return True
 
     def _eval_is_upper_hessenberg(self):
-        return all(self[i, j].is_zero
+        return self.is_square and all(self[i, j].is_zero
                    for i in range(2, self.rows)
                    for j in range(min(self.cols, (i - 1))))
```

Comment:
PEP-8 requires indentation to be aligned with the open brackets. However, I think how you have it now is fine, since SymPy is fairly liberal when it comes to that sort of indentation.